builder/vsphere/driver/vm.go

```diff
@@ -625,6 +626,28 @@ func (vm *VirtualMachineDriver) Configure(config *HardwareConfig) error {
 	}
 
 	_, err = task.WaitForResult(vm.driver.ctx, nil)
+	if err != nil {
+		return err
+	}
+
+	devices, err := vm.Devices()
+	if err != nil {
+		return err
+	}
+	TPMs := devices.SelectByType((*types.VirtualTPM)(nil))
+	hasTPM := len(TPMs) > 0
+	if config.VTPMEnabled != hasTPM {
+		if !hasTPM {
+			device := &types.VirtualTPM{}
+			err = vm.addDevice(device)
+		} else {
+			err = vm.RemoveDevice(false, TPMs...)
```

Comment:
Is this important to add too ?

Reply:
AFAIR I've copy-pasted idea of removing existing TPM from somewhere else. Not sure whether if removing TPM should be supported. I'd vote for keeping that code.

Reply:
Cool, you probably have much more insight in this code area than me, let's keep this !